GG/src/GUI.cpp

```diff
@@ -296,16 +303,17 @@ GUIImpl::GUIImpl() :
     m_save_as_png_wnd(nullptr),
     m_clipboard_text()
 {
-    m_drag_wnds[0] = m_drag_wnds[1] = m_drag_wnds[2] = nullptr;
+    m_drag_wnds[0].reset();
+    m_drag_wnds[1].reset();
+    m_drag_wnds[3].reset();
     m_mouse_button_state[0] = m_mouse_button_state[1] = m_mouse_button_state[2] = false;
 }
 
 void GUIImpl::HandleMouseButtonPress(unsigned int mouse_button, const Pt& pos, int curr_ticks)
 {
-    m_curr_wnd_under_cursor = GUI::s_gui->CheckedGetWindowUnder(pos, m_mod_keys);
+    auto curr_wnd_under_cursor = GUI::s_gui->CheckedGetWindowUnder(pos, m_mod_keys);
```

Comment:
What is `curr_wnd_under_cursor` doing that assigning directly to `m_curr_wnd_under_cursor`  can't?